src/Silex/Application.php

```diff
@@ -73,7 +76,7 @@ public function __construct(array $values = array())
         };
 
         $this['controllers_factory'] = $this->factory(function () use ($app) {
-            return new ControllerCollection($app['route_factory']);
+            return new ControllerCollection($app, $app['route_factory']);
```

Comment:
I would make the new argument second instead of prepending it to avoid an unneeded BC break. I would also inject the routes factory instead of the whole app.